services/validators.go

```diff
@@ -31,6 +31,18 @@ func ValidateJob(j models.JobSpec, store *store.Store) error {
 	return merr
 }
 
+// ValidateAdaptor checks that the bridge type doesn't have a duplicate name
+func ValidateAdaptor(bt *models.BridgeType, store *store.Store) error {
+	var err error
+	ts := models.TaskSpec{Type: bt.Name}
+	if _, err = adapters.For(ts, store); err == nil {
+		err = fmt.Errorf("adaptor validation: adaptor %v exists", bt.Name)
+	} else if strings.Contains(err.Error(), "not a supported adapter type") {
```

Comment:
This else statement is really hacky...

Reply:
Updated it to check whether the adaptor is nil or not. Much cleaner.